### PR TITLE
[WIP] target .NET Standard 1.0

### PR DESCRIPTION
**Pull request by blairconrad**
Very much a work in progress, and I've not had time in the last week or so to look at it, but figured we might as well have a place to chat about the work as we can.

I've adopted a style much like xunit's - `xprojs` everywhere for the multi-targeting. Essentially I'm spiking a 4.5 build created using the `dotnet` tooling.

Points of interest:
- I renamed the AcceptanceTests directory, because once they work, they won't be just .NET 4.5 anymore
- `SpySink` is using the `LongLivedMarshallByRefObject` from `Xunit.Sdk`. There's also 2 (!) in `Xunit`, and I picked the `Xunit.Sdk` one for now because I couldn't figure out how to do an extern alias using the `dotnet` tooling. I fear there's no way. I am not sure whether that's the cause of the next point.
- the acceptance tests build, but when run only find the single `Fact`
- and to even get that to work, I had to have the tests target .NET 4.5.1, as that's the lowest version `dotnet-test-xunit` supports. Again, maybe a factor in the previous failure.

Thanks! I'm having a great time.

Connects to #300

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +7,5 @@ obj
 Thumbs.db
 *.sln.ide
 packages
+*.nuget.targets
+project.lock.json
```
Comment (blairconrad) [inline .gitignore]: I see it. Drat.

**Comment by blairconrad** [comment]
Also, now that I see the diff in this interface, I notice a few extraneous changes (jiggled lines in the .sln, package upgrades in the 1.0 projects). Of course I'd fix all of those if ever we got to the point of a mergeableish branch.

**Comment by adamralph** [comment]
@blairconrad this is awesome work, thanks!

I'll cast an eye on it ASAP.

**Comment by blairconrad** [comment]
In case we're keeping track, I've not yet had time to come back and figure out why the `[Scenario]`s aren't being found. And the next 10–11 days don't look great either, but I'm happy to return if it's not resolved by then!

**Comment by adamralph** [comment]
@blairconrad I've done the things I mentioned, which I would hope make switching to netstd, xproj, etc. easier. I.e. dropping the 1.x projects, switching to AppVeyor and switching from StyleCop.MSBuild to StyleCopAnalyzers.

Unfortunately it means this needs a rebase. I'm happy to take care of it, but would you like the honour?

**Comment by blairconrad** [comment]
@adamralph, I'm glad to take a stab.
It's... not as straightforward as I'd thought. Getting a confusing conflict with the two `XUnit.LongLivedMarshallByRefObject`s in the newer XUnit packages. 
Odd thing is that I'm getting the conflict at commit "Upgrade xunit to latest pre-releases, where needed", which I did not before.
Bedtime now, but I can continue in the morning, if you're in no rush. Otherwise, I will not be offended if you rebase. It's all for the good of xBehave.

Also, `xunit.core.2.1.0\build\portable-net45+win8+wp8+wpa81\xunit.core.props` doesn't exist in the newer packages. Considering omitting, but TBH, I don't know what the props files are supposed to do, so amn't sure if this is a big deal.

**Comment by adamralph** [comment]
@blairconrad there's no huge rush. Happy to wait until you have time.

> Also, xunit.core.2.1.0\build\portable-net45+win8+wp8+wpa81\xunit.core.props doesn't exist in the newer packages. Considering omitting, but TBH, I don't know what the props files are supposed to do, so amn't sure if this is a big deal.

If xunit doesn't need it then xbehave doesn't either. AFAIK xunit has dropped the PCL x-plat support in favour of netstd so xbehave should do the same (this may be a naive comment since I haven't looked into the details yet).

**Comment by blairconrad** [comment]
Hey, @adamralph. Rebased.
Well, with the changes that you made, it was more like "did again from scratch".
Regrettably, nothing works any better than I initially reported. But we're no farther behind, and we're rebased!

I didn't begin to touch the build script, but a `dotnet restore` at the root will restore everything.
Then a `dotnet build -c Release` in each of the 3 projects (in order) will build them.

`dotnet test` in `Xbehave.Test.Acceptance.Net45` will find one test only (the one with the `Fact`) and fail.

**Comment by blairconrad** [comment]
That's as far as I go tonight. Tomorrow morning's work will be the FakeItEasy release, if nobody else has done it.
After that, I'm yours again. But currently completely at sea about the missing tests.

**Comment by adamralph** [comment]
@blairconrad thanks very much! I'll take a look...

**Comment by adamralph** [comment]
@blairconrad I managed to get everything working. The problem was that the output names for Xbehave.Execution were not set properly. I'll send a superseding PR shortly with an extra commit.

**Comment by blairconrad** [comment]
Oh, wow! Go you! :tada:

I feel bad for not doing it right. And for not getting back to it.
Turns out on Monday a severe LibraryHippo issue was reported and that sucked up quite a lot of my funtime. Also sleepingtime.

**Comment by adamralph** [comment]
@blairconrad your commits were merged as part of #321. Thanks again for your help with this!

**Comment by blairconrad** [comment]
> @blairconrad your commits were merged as part of #321. Thanks again for your help with this!

A pleasure. I'm sorry I never returned to finish the work. I'm an unreliable helper!

**Pull request closed without merging**